cli/tests/test_completion.rs

```diff
@@ -78,15 +160,12 @@ fn test_bookmark_names() {
     origin_dir.run_jj(["git", "export"]).success();
     work_dir.run_jj(["git", "fetch"]).success();
 
-    let mut test_env = test_env;
     // Every shell hook is a little different, e.g. the zsh hooks add some
     // additional environment variables. But this is irrelevant for the purpose
     // of testing our own logic, so it's fine to test a single shell only.
-    test_env.add_env_var("COMPLETE", "fish");
-    let test_env = test_env;
     let work_dir = test_env.work_dir("repo");
 
-    let output = work_dir.run_jj(["--", "jj", "bookmark", "rename", ""]);
+    let output = work_dir.complete_at(Shell::Fish, 3, &["jj", "bookmark", "rename", ""]);
```

Comment:
nits:
* `&` can be omitted
* might be good to omit `"jj", ` too.
* add `.complete(shell, args)` (without `index`) or `.complete_fish(args)` for convenience?

Reply:
`&` was actually intentional to avoid monomorphizing the generic for each length, but doing so would throw the design as a generic API as a whole into question, so I've changed it now to pass the arrays by value. Might be interesting to measure compile times...